src/components/RefinementList/RefinementList.js

```diff
@@ -127,6 +128,8 @@ class RefinementList extends React.Component {
     }
 
     const limit = this.state.isShowMoreOpen ? this.props.limitMax : this.props.limitMin;
+    this.props.facetValues = this.props.facetValues.slice(0, limit);
```

Comment:
this is not doable in React, you need to get a new reference instead of modifying props